evm_arithmetization/src/witness/transition.rs

```diff
@@ -276,41 +331,64 @@ fn perform_op<F: Field>(
         }
         Operation::BinaryArithmetic(op) => generate_binary_arithmetic_op(op, state, row)?,
```

Comment:
This means that the interpreter is also pushing to the logic stark?

Reply:
Instead of having an if/else in each `generate_X` function, we thought it would be cleaner to always generate the rows (whether we are in the interpreter or not) and then clear all STARKs but the memory STARK at the end of the operation. This way, even if we generate rows, we still don't store them on the interpreter side.

Reply:
So the difference between Interpreter's generation state and the one used for witness generation is just that the former doesn't push to the cpu?

Reply:
They both push all rows, but the rows are immediately cleared on the interpreter's side.